engine/src/connector.rs

```diff
@@ -62,613 +52,277 @@ impl<I: IO + Copy> EthConnectorContract<I> {
+            method: "engine_ft_transfer".to_string(),
+            args: data,
+            attached_balance: Yocto::new(1),
+            attached_gas: GAS_FOR_FT_TRANSFER,
+        }
     }
 
     /// FT transfer call from sender account (invoker account) to receiver
     /// We starting early checking for message data to avoid `ft_on_transfer` call panics
     /// But we don't check relayer exists. If relayer doesn't exist we simply not mint/burn the amount of the fee
     /// We allow empty messages for cases when `receiver_id =! current_account_id`
-    pub fn ft_transfer_call(
-        &mut self,
-        predecessor_account_id: AccountId,
-        current_account_id: AccountId,
-        args: TransferCallCallArgs,
-        prepaid_gas: NearGas,
-    ) -> Result<PromiseWithCallbackArgs, error::FtTransferCallError> {
-        sdk::log!(
-            "Transfer call to {} amount {}",
-            args.receiver_id,
-            args.amount,
-        );
-
-        // Verify message data before `ft_on_transfer` call to avoid verification panics
-        // It's allowed empty message if `receiver_id =! current_account_id`
-        if args.receiver_id == current_account_id {
-            let message_data = FtTransferMessageData::parse_on_transfer_message(&args.msg)
-                .map_err(error::FtTransferCallError::MessageParseFailed)?;
-            // Check is transfer amount > fee
-            if message_data.fee.as_u128() >= args.amount.as_u128() {
-                return Err(error::FtTransferCallError::InsufficientAmountForFee);
-            }
-
-            // Additional check overflow before process `ft_on_transfer`
-            // But don't check overflow for relayer
-            // Note: It can't overflow because the total supply doesn't change during transfer.
-            let amount_for_check = self
-                .ft
-                .internal_unwrap_balance_of_eth_on_aurora(&message_data.recipient);
-            if amount_for_check
-                .checked_add(Wei::from(args.amount))
-                .is_none()
-            {
-                return Err(error::FtTransferCallError::Transfer(
-                    fungible_token::error::TransferError::BalanceOverflow,
-                ));
-            }
-            if self
-                .ft
-                .total_eth_supply_on_aurora
-                .checked_add(Wei::from(args.amount))
-                .is_none()
-            {
-                return Err(error::FtTransferCallError::Transfer(
-                    fungible_token::error::TransferError::TotalSupplyOverflow,
-                ));
-            }
+    pub fn ft_transfer_call(&self, data: Vec<u8>) -> PromiseCreateArgs {
+        PromiseCreateArgs {
+            target_account_id: self.get_eth_connector_contract_account(),
+            method: "engine_ft_transfer_call".to_string(),
+            args: data,
+            attached_balance: Yocto::new(1),
+            attached_gas: GAS_FOR_FT_TRANSFER_CALL,
```

Comment:
It is unpredictable how much gas the ft_transfer_call cost because it depends on the `ft_on_transfer` function. But here we provide the constant gas amount and users don't have the opportunity to increase it.

What about using the trick from the standard realization: https://docs.rs/near-contract-standards/latest/src/near_contract_standards/fungible_token/core_impl.rs.html#146 ?